state/core_access.go

```diff
@@ -402,13 +403,27 @@ func (ca *CoreAccessor) SubmitTxWithBroadcastMode(
 	tx Tx,
 	mode sdktx.BroadcastMode,
 ) (*TxResponse, error) {
-	txResp, err := apptypes.BroadcastTx(ctx, ca.coreConn, mode, tx)
+	txResp, err := broadcastTx(ctx, ca.coreConn, mode, tx)
 	if err != nil {
 		return nil, err
 	}
 	return unsetTx(txResp.TxResponse), nil
 }
 
+// broadcastTx uses the provided grpc connection to broadcast a signed and
+// encoded transaction.
+func broadcastTx(ctx context.Context, conn *grpc.ClientConn, mode sdktx.BroadcastMode, txBytes []byte) (*sdktx.BroadcastTxResponse, error) {
+	txClient := sdktx.NewServiceClient(conn)
+
+	return txClient.BroadcastTx(
+		ctx,
+		&sdktx.BroadcastTxRequest{
+			Mode:    mode,
+			TxBytes: txBytes,
+		},
+	)
+}
```

Comment:
It looks like there was a [BroadcastTx](https://github.com/celestiaorg/celestia-app/blob/21b5bc747c8500e4888474df7d828e66c33f332d/x/blob/types/builder.go#L252-L263) on v1.x that no longer exists on celestia-app v2.x so I copied it here. I couldn't use the signer `BroadcastTx` because that accepts SDK txs and this function accepts txBytes as a byte slice.

cc: @cmwaters if you have other ideas.